# Matriz

$\textit{def}$($\textbf{Matriz}$): es un arreglo rectangular de números dispuestos en renglones y columnas, ejemplo:
$$
M = \left( \begin{array}{ccc}
a & b & c \\
d & e & f \\
g & h & i \\
j & k & l\end{array} \right)
$$
En este ejemplo es una matriz de nombre $M$ que tiene 4 renglones y 3 columnas; donde el elemento $m_{11} = a$; y el elemento $m_{43}=l$

La dimensión de la matriz se denota de la siguiente manera $M \in \Re^{4*3} $


Se usa nombrar con mayúsculas a las matrices; pero cuando se hace referencia a un elemento de la matriz, se escribe el nombre de la matrix en minúscula y 2 subíndices:
primero el correspondiente al renglón y después el de columna. Por ejemplo, el elemento $f$ de la matrix $M$  viene siendo $m_{1,2}$.

ver: https://docs.scipy.org/doc/numpy-1.10.4/reference/routines.linalg.html


In [4]:
import numpy as np
A=np.array([[1,2,3],[4,5,6],[7,8,9],[10,11,12]])
print('A= ',A,sep='\n',end='\n\n')
print('Numero de dimensiones= ',A.ndim,end='\n\n') 
print('Forma de la dimension= ',A.shape,end='\n\n')
print('Tamaño= ',A.size,end='\n\n')
print('Tipo de los datos= ',A.dtype,end='\n\n')
print('A[:,1:3]= ',A[:,1:3],sep='\n',end='\n\n')
print('A[1:3,:]= ',A[1:3,:],sep='\n',end='\n\n')
A = np.array([[1, 2], [3, 4]])
B = np.array([[5, 6]])
print('A= ',A,sep='\n',end='\n\n')
print('B= ',B,sep='\n',end='\n\n')
C = np.concatenate((A, B), axis=0)
print('A concatenada con B en eje 0= ', C,sep='\n',end='\n\n')
C = np.concatenate((A, B.T), axis=1)
print('A concatenada con B en eje 1= ', C,sep='\n',end='\n\n')
C = np.concatenate((A, B), axis=None)
print('A concatenada con B en ningún eje= ', C,sep='\n',end='\n\n')

B=np.zeros([2,3],int)
print('zeros[2,3] int= ',B,sep='\n',end='\n\n')
B=np.zeros([2,3],float)
print('zeros[2,3] float= ',B,sep='\n',end='\n\n')
B=np.ones([2,3],int)
print('ones[2,3] int= ',B,sep='\n',end='\n\n')
#arange([start,]stop[,step,],dtype=None). Crea un array con valores distanciados step entre el valor inicial star y el valor final stop. Si no se establece step python establecerá uno por defecto.
B=np.arange(0,10,.5)
print('arange(0,10,.5)= ',B,sep='\n',end='\n\n')
#linspace(start,stop,num,endpoint=True,retstep=False). Crea un array con valor inicial start, valor final stop y num elementos
B=np.linspace(0,10,20)
print('linspace(0,10,50)= ',B,sep='\n',end='\n\n')
#meshgrid(x,y). Genera una matriz de coordenadas a partir de dos los arrays x, y.
(X,Y)=np.meshgrid([0,1,2],[3,4,5,6])
print('X de meshgrid([0,1,2],[3,4,5,6])= ',X,sep='\n',end='\n\n')
print('Y de meshgrid([0,1,2],[3,4,5,6])= ',Y,sep='\n',end='\n\n')

A= 
[[ 1  2  3]
 [ 4  5  6]
 [ 7  8  9]
 [10 11 12]]

Numero de dimensiones=  2

Forma de la dimension=  (4, 3)

Tamaño=  12

Tipo de los datos=  int64

A[:,1:3]= 
[[ 2  3]
 [ 5  6]
 [ 8  9]
 [11 12]]

A[1:3,:]= 
[[4 5 6]
 [7 8 9]]

A= 
[[1 2]
 [3 4]]

B= 
[[5 6]]

A concatenada con B en eje 0= 
[[1 2]
 [3 4]
 [5 6]]

A concatenada con B en eje 1= 
[[1 2 5]
 [3 4 6]]

A concatenada con B en ningún eje= 
[1 2 3 4 5 6]

zeros[2,3] int= 
[[0 0 0]
 [0 0 0]]

zeros[2,3] float= 
[[0. 0. 0.]
 [0. 0. 0.]]

ones[2,3] int= 
[[1 1 1]
 [1 1 1]]

arange(0,10,.5)= 
[0.  0.5 1.  1.5 2.  2.5 3.  3.5 4.  4.5 5.  5.5 6.  6.5 7.  7.5 8.  8.5
 9.  9.5]

linspace(0,10,50)= 
[ 0.          0.52631579  1.05263158  1.57894737  2.10526316  2.63157895
  3.15789474  3.68421053  4.21052632  4.73684211  5.26315789  5.78947368
  6.31578947  6.84210526  7.36842105  7.89473684  8.42105263  8.94736842
  9.47368421 10.        ]

X de meshgrid([0,1,2],[3,4,5,6])= 
[[0 1 2]
 [0 1 2]
 [0 1 2]
 [0 1 2]]

Y de meshgrid([0,1,

# Matriz Identidad

La matriz identidad es una matriz que cumple la propiedad de ser el elemento neutro del producto de matrices.
Esto quiere decir que el producto de cualquier matriz por la matriz identidad no tiene ningún efecto.

Como el producto de matrices sólo tiene sentido si sus dimensiones son compatibles,
existen infinitas matrices identidad dependiendo de las dimensiones.

$I_{n}$ es la matriz identidad de tamaño $n$, y se define como la matriz diagonal que tiene valor 1 en cada una de las entradas de la diagonal principal, y 0 en el resto. Así por ejemplo:

$I_{1}=\left( \begin{array}{c}1\end{array} \right)$,

$I_{2}=\left( \begin{array}{cc}1&0\\0&1\end{array} \right)$,

$I_{3}=\left( \begin{array}{ccc}1&0&0\\0&1&0\\0&0&1\end{array} \right)$,

$...$,

$I_{n}=\left( \begin{array}{cccc}1&0&...&0\\0&1&...&0\\\vdots&\vdots&\ddots&\vdots\\0&0&...&1\end{array} \right)$


In [4]:
import numpy as np
A=np.eye(3)
print(A)

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]


# Matriz Inversa

La inversa de la matriz $A$, es otra matriz, del mismo orden que $A$, tal que al efectuar el producto, se obtiene como resultado la matriz identidad.

Sólo las matrices cuadradas pueden tener inversa, sin embargo, el hecho de que una matriz sea cuadrada no garantiza la existencia de una matriz inversa.


si A es una matriz m x m, y y si ésta tiene inversa:


$A \cdot  A^{-1} = A^{-1} \cdot A= I$


Ejemplo:

$\left( \begin{array}{cc}3&4\\2&16\end{array} \right) \left( \begin{array}{cc}0.4&-0.1\\-0.05&0.075\end{array} \right) = \left( \begin{array}{cc}1&0\\0&1\end{array} \right) = I_2$


In [6]:
import numpy as np
A=np.array([[3,4],[2,16]])
print('A= ',A,sep='\n',end='\n\n')
B=np.linalg.inv(A)
print('inv(A)= ',B,sep='\n',end='\n\n')
print('(A)(inv(A))= ',A.dot(B),sep='\n',end='\n\n')
print('(inv(A))(A)= ',B.dot(A),sep='\n',end='\n\n')

A= 
[[ 3  4]
 [ 2 16]]

inv(A)= 
[[ 0.4   -0.1  ]
 [-0.05   0.075]]

(A)(inv(A))= 
[[1. 0.]
 [0. 1.]]

(inv(A))(A)= 
[[1. 0.]
 [0. 1.]]



# Determinante de una Matriz

El determinante de una matriz $A$ es un escalar único asociado con los elementos de una matriz, por medio de operaciones definidas.

El determinante se simboliza por: det($A$)    ó bien, envolviendo la matriz con barras verticales $|A|$

El valor de un determinante sólo se obtiene en matrices cuadradas.

Existen varios métodos para encontrar determinantes, para matrices de $2^{o}$ y $3^{o}$ orden se usa el método de diagonales que consiste en:

$|A|=\left( \begin{array}{cc} a_{00}&a_{01}\\a_{10}&a_{11}\end{array} \right)=a_{00}a_{11}-a_{10}a_{01}=det(A)$

$|A|=\left( \begin{array}{ccc} a_{00}&a_{01}&a_{02}\\a_{10}&a_{11}&a_{12}\\a_{20}&a_{21}&a_{22}\end{array} \right)=a_{00}a_{11}a_{22}+a_{01}a_{12}a_{20}+a_{10}a_{21}a_{02}-a_{02}a_{11}a_{20}-a_{00}a_{12}a_{21}-a_{01}a_{10}a_{22}=det(A)$

Para matrices mayores se puede aplicar el método de cofactores, siempre y cuando sea una matriz cuadrada.

el det(A$\cdot$B) = det(A) $\cdot$ det(B), es decir: 

$|A\cdot B|=|A|\cdot|B|$

In [45]:
import numpy as np
A=np.array([[3,4],[2,16]])
print('A= ',A,sep='\n',end='\n\n')
detA= np.linalg.det(A)
print('det(A)= ',detA,sep='\n',end='\n\n')
B=np.array([[5,7],[3,8]])
print('B= ',B,sep='\n',end='\n\n')
detB= np.linalg.det(B)
print('det(B)= ',detB,sep='\n',end='\n\n')
AB=A.dot(B)
print('AB= ',AB,sep='\n',end='\n\n')
detAB= np.linalg.det(AB)
print('det(AB)= ',detAB,sep='\n',end='\n\n')
print("({0})  ({1}) = {2}".format(detA,detB,detA*detB))

A= 
[[ 3  4]
 [ 2 16]]

det(A)= 
40.000000000000014

B= 
[[5 7]
 [3 8]]

det(B)= 
18.999999999999996

AB= 
[[ 27  53]
 [ 58 142]]

det(AB)= 
759.9999999999994

(40.000000000000014)  (18.999999999999996) = 760.0000000000001


# Matriz Regular

Es aquella que si tiene matriz inversa, ya que su determinante es diferente de cero.

In [38]:
import numpy as np
A=np.array([[3,4,5],[4,16,8],[7,6,9]])
print('A= ',A,sep='\n',end='\n\n')
detA=np.linalg.det(A)
print('det(A)= ',detA,sep='\n',end='\n\n')

A= 
[[ 3  4  5]
 [ 4 16  8]
 [ 7  6  9]]

det(A)= 
-71.99999999999994



# Matriz Singular

Es aquella matriz que no tiene inversa porque su determinante es igual a cero, generalmente es cuando una columna/renglón es múltiplo de otra columna/renglón.

In [7]:
import numpy as np
A=np.array([[3,4,5],[4,16,8],[2,8,4]])
print('A= ',A,sep='\n',end='\n\n')
detA=np.linalg.det(A)
print('det(A)= ',detA,sep='\n',end='\n\n')
B=np.array([[3,4,6],[4,16,8],[6,8,12]])
print('B= ',B,sep='\n',end='\n\n')
detB=np.linalg.det(B)
print('det(B)= ',detB,sep='\n',end='\n\n')

A= 
[[ 3  4  5]
 [ 4 16  8]
 [ 2  8  4]]

det(A)= 
0.0

B= 
[[ 3  4  6]
 [ 4 16  8]
 [ 6  8 12]]

det(B)= 
0.0



# Matriz Cofactor

La matriz cofactor se representa por $A^{C}$ y esta formada por los cofactores de cada elemento de la matriz original.

Sólo las matrices cuadradas tienen matriz cofactor, la fórmula para obtener cofactores es: $\alpha_{ij}=(-1)^{i+j}M_{i,j}$

$M_{i,j}$ .- Es un determinante de menor orden que la original y se obtiene al eliminar la fila y la columna del elemento con que se esta trabajando.

Ejemplo:
$A=\left( \begin{array}{ccc}
a_{1,1}&a_{1,2}&a_{1,3}\\
a_{2,1}&a_{2,2}&a_{2,3}\\
a_{3,1}&a_{3,2}&a_{3,3}
\end{array} \right)$,

$A^C=\left( \begin{array}{ccc}
\alpha_{1,1}&\alpha_{1,2}&\alpha_{1,3}\\
\alpha_{2,1}&\alpha_{2,2}&\alpha_{2,3}\\
\alpha_{3,1}&\alpha_{3,2}&\alpha_{3,3}
\end{array} \right)$

donde por ejemplo: $\alpha_{3,3}=(-1)^{3+3}\left| \begin{array}{cc} a_{11}&a_{12}\\a_{21}&a_{22}\end{array} \right|=(1)*(a_{11}a_{22}-a_{21}a_{12})$

$A^C=
   \begin{pmatrix} 
      +
      \left |
         \begin{matrix} a_{22} & a_{23} \\
             a_{32} & a_{33} 
         \end{matrix}
      \right | &
      -
      \left |
         \begin{matrix}
            a_{21} & a_{23} \\
            a_{31} & a_{33}
         \end{matrix}
      \right | &
      +
      \left |
         \begin{matrix}
            a_{21} & a_{22} \\
            a_{31} & a_{32}
         \end{matrix}
      \right | \\
       & & \\
      -
      \left |
         \begin{matrix}
            a_{12} & a_{13} \\
            a_{32} & a_{33}
         \end{matrix}
      \right | &
      +
      \left |
         \begin{matrix}
            a_{11} & a_{13} \\
            a_{31} & a_{33}
         \end{matrix}
      \right | &
      -
      \left |
         \begin{matrix}
            a_{11} & a_{12} \\
            a_{31} & a_{32}
         \end{matrix}
      \right| \\
      & & \\
      +
      \left |
         \begin{matrix}
            a_{12} & a_{13} \\
            a_{22} & a_{23}
         \end{matrix}
      \right | &
      -
      \left |
         \begin{matrix}
            a_{11} & a_{13} \\
            a_{21} & a_{23}
         \end{matrix}
      \right | &
      +
      \left |
         \begin{matrix}
            a_{11} & a_{12} \\
            a_{21} & a_{22}
         \end{matrix}
      \right|
   \end{pmatrix} =
   \begin{pmatrix} 
      a_{22}a_{33} - a_{23}a_{32} & a_{23}a_{31} - a_{21}a_{33} & a_{21}a_{32} - a_{22}a_{31}\\
      a_{32}a_{13} - a_{33}a_{12} & a_{33}a_{11} - a_{31}a_{13} & a_{31}a_{12} - a_{32}a_{11}\\
      a_{12}a_{23} - a_{13}a_{22} & a_{13}a_{21} - a_{11}a_{23} & a_{11}a_{22} - a_{12}a_{21}
   \end{pmatrix}
$

In [8]:
import numpy as np
A=np.array([[3,4,5],[4,16,8],[2,8,4]])
print('A= ',A,sep='\n',end='\n\n')

def matrix_cofactor(matrix): 
    C = np.zeros(matrix.shape) 
    (nrows, ncols) = C.shape 
    minor = np.zeros([nrows-1, ncols-1])
    for row in range(nrows): 
        for col in range(ncols): 
            minor[:row,:col] = matrix[:row,:col] 
            minor[row:,:col] = matrix[row+1:,:col] 
            minor[:row,col:] = matrix[:row,col+1:] 
            minor[row:,col:] = matrix[row+1:,col+1:] 
            C[row, col] = (-1)**(row+col) * np.linalg.det(minor) 
    return C 

cofA=matrix_cofactor(A)
print('cofactor(A)= ',cofA,sep='\n',end='\n\n')



A= 
[[ 3  4  5]
 [ 4 16  8]
 [ 2  8  4]]

cofactor(A)= 
[[  0.  -0.   0.]
 [ 24.   2. -16.]
 [-48.  -4.  32.]]



# Matriz Transpuesta

Sea $A$ una matriz. La matriz transpuesta, denotada con $A^{t}$ está dada por aquella matriz donde el elemento $a_{ji}$ de la matriz original A se convertirá en el elemento $a_{ij}$ de la matriz transpuesta $A^t$.

In [86]:
import numpy as np
A=np.array([[3,4,5,7],[4,16,8,2],[2,8,4,0]])
print('A= ',A,sep='\n',end='\n\n')
transpuestaA=A.T
print('transpuesta(A)= ',transpuestaA,sep='\n',end='\n\n')


A= 
[[ 3  4  5  7]
 [ 4 16  8  2]
 [ 2  8  4  0]]

transpuesta(A)= 
[[ 3  4  2]
 [ 4 16  8]
 [ 5  8  4]
 [ 7  2  0]]



# Matriz Adjunta

La matriz adjunta es la transpuesta de la matriz cofactor y se representa por ADJ A.  

Su fórmula es: $ADJ\;A=(A^{C})^{T}$

# Vectores

Los vectores son un caso especial de matrices donde una de sus dimensiones es 1; de esta forma tenemos vector renglón, y vector columna.


Ejemplo de vector renglón: $X = \left( \begin{array}{ccccc}m & n & o & p & q \end{array} \right)$

Ejemplo de vector columna: $Y = \left( \begin{array}{c}r \\s\\t\\u\end{array} \right)$

La dimensión de un vector se denota de la siguiente manera $X \in \Re^{5}; \;\; Y\in \Re^{4}$

Cuando se hace referencia a un elemento del vector, se escribe el nombre del vector y 1 subíndice.Por ejemplo, el elemento $p$ del vector $X$ viene siendo $X[3]=x_{3}$; o el elemento $s$ del vector $Y$ viene siendo $Y[1]=y_{1}$


# Suma y Resta de matrices

La suma y resta de matrices solo se puede realizar cuando las 2 matrices tienen la misma dimensión,
y la suma o resta se realiza con los elementos correspondientes, es decir, los elementos que tienen los mismos subíndices, ejemplo:

$A=\left( \begin{array}{cc}a_{1,1}&a_{1,2}\\a_{2,1}&a_{2,2}\end{array} \right)$,

$B=\left( \begin{array}{cc}b_{1,1}&b_{1,2}\\b_{2,1}&b_{2,2}\end{array} \right)$

$A+B=\left( \begin{array}{cc}a_{1,1}+b_{1,1}&a_{1,2}+b_{1,2}\\a_{2,1}+b_{2,1}&a_{2,2}+b_{2,2}\end{array} \right)$



In [124]:
A = np.array([[1, 2], [3, 4]])
B = np.array([[5, 6], [7, 8]])
print('A= ',A,sep='\n',end='\n\n')
print('B= ',B,sep='\n',end='\n\n')
C = A+B
print('A + B = ', C,sep='\n',end='\n\n')

A= 
[[1 2]
 [3 4]]

B= 
[[5 6]
 [7 8]]

A + B = 
[[ 6  8]
 [10 12]]



# Multiplicación de matrices

La multiplicación de matrices solo se puede realizar cuando la segunda dimensión de la primer matriz es igual a la primera dimensión de la segunda matriz, y la matriz resultante tiene como dimensiones la primera dimensión de la primer matriz y la segunda dimensión de la segunda matriz; en otras palabras, suponiendo $A \in \Re^{M*N}$ y $B \in \Re^{N*P}$; $(A*B) \in \Re^{M*P}$

De esta forma, el elemento $(A*B)_{ij}$ se forma al multiplicar el renglón $i$ de la matriz $A$, por la columna $j$ de la matriz $B$.

Ejemplo:

$A=\left( \begin{array}{cc}a_{1,1}&a_{1,2}\\a_{2,1}&a_{2,2}\end{array} \right)$,

$B=\left( \begin{array}{cc}b_{1,1}&b_{1,2}\\b_{2,1}&b_{2,2}\end{array} \right)$

$(A*B)=\left( \begin{array}{cc}a_{1,1}\cdot b_{1,1}+a_{1,2}\cdot b_{2,1} & a_{1,1}\cdot b_{1,2}+a_{1,2}\cdot b_{2,2}\\
a_{2,1}\cdot b_{1,1}+a_{2,2}\cdot b_{2,1} & a_{2,1}\cdot b_{1,2}+a_{2,2}\cdot b_{2,2}\end{array} \right)$

In [10]:
A = np.array([[1, 2], [3, 4]])
B = np.array([[5, 6], [7, 8]])
print('A= ',A,sep='\n',end='\n\n')
print('B= ',B,sep='\n',end='\n\n')
C = A.dot(B)
print('A * B = ', C,sep='\n',end='\n\n')
print('A * B = ', A@B,sep='\n',end='\n\n')

A= 
[[1 2]
 [3 4]]

B= 
[[5 6]
 [7 8]]

A * B = 
[[19 22]
 [43 50]]

A * B = 
[[19 22]
 [43 50]]



# División de matrices

La división de matrices no existe, se usa la multiplicación por la matriz inversa que es lo más parecido a dividir en el mundo de las matrices.

# Multiplicación de una matriz por un escalar

Cada elemento de la matriz se multiplica de manera individual por el escalar, ejemplo:

$A=\left( \begin{array}{cc}2&2\\4&6\end{array} \right)$

$3*A=\left( \begin{array}{cc}3*2&3*2\\3*4&3*6\end{array} \right)$

$3*A=\left( \begin{array}{cc}6&6\\12&18\end{array} \right)$

In [127]:
A = np.array([[2, 2], [4, 6]])
print('A= ',A,sep='\n',end='\n\n')
print('3A= ',3*A,sep='\n',end='\n\n')

A= 
[[2 2]
 [4 6]]

3A= 
[[ 6  6]
 [12 18]]



# División de una matriz por un escalar

Cada elemento de la matriz se multiplica de manera individual por el reciproco del escalar, ejemplo:

$A=\left( \begin{array}{cc}2&2\\4&6\end{array} \right)$

$\frac{A}{3}=\frac{1}{3}*A=\left( \begin{array}{cc}\frac{1}{3}*2&\frac{1}{3}*2\\\frac{1}{3}*4&\frac{1}{3}*6\end{array} \right)$

$\frac{1}{3}A=\left( \begin{array}{cc}\frac{2}{3}&\frac{2}{3}\\\frac{4}{3}&2\end{array} \right)$

In [128]:
A = np.array([[2, 2], [4, 6]])
print('A= ',A,sep='\n',end='\n\n')
print('3A= ',1/3*A,sep='\n',end='\n\n')

A= 
[[2 2]
 [4 6]]

3A= 
[[0.66666667 0.66666667]
 [1.33333333 2.        ]]



# Ecuación Matricial

Una ecuación matricial es una ecuación que podemos expresar por medio de matrices, o bien una ecuación en la que intervienen matrices.

$\textbf{Ejemplo 1}$

Resuelva (obtenga $X$ ) la siguiente ecuación matricial $A + B – X = C$, donde A, B, C y X son matrices.

$A = \left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)$ $B = \left( \begin{array}{cc}6 & 5\\2 & 3\end{array} \right)$
$C = \left( \begin{array}{cc} 5 & 6\\3 & 5\end{array} \right)$

$A+B-X=C$

$\left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)+\left( \begin{array}{cc}6 & 5\\2 & 3\end{array} \right)-X=\left( \begin{array}{cc} 5 & 6\\3 & 5\end{array} \right)$

Despejando..: $X=A+B-C$

$X=\left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)+\left( \begin{array}{cc}6 & 5\\2 & 3\end{array} \right)-\left( \begin{array}{cc} 5 & 6\\3 & 5\end{array} \right)$

$X = \left( \begin{array}{cc}6 & 3\\2 & 6\end{array} \right)$

$\textbf{Ejemplo 2}$

Resuelva X de la ecuación matricial $AX+B=3X$, donde:

$A = \left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)$ $B = \left( \begin{array}{ccc}4 & 0 & -2\\1 & -3 & -1\end{array} \right)$

Para que $AX$ se pueda sumar a $B$, la multiplicación de $AX$ debe arrojar una matriz de $2 * 3$. Esto se obtiene si $X$ tiene 2 filas y 3 columnas.

$AX+B=3X$

$\left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)X+\left( \begin{array}{ccc}4 & 0 & -2\\1 & -3 & -1\end{array} \right)=3\left( \begin{array}{cc}1 & 0\\0 & 1\end{array} \right)X$

Despejando..: $(A-3)X=-B$

$\left[ \left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)-3\left(\begin{array}{cc}1 & 0\\0 & 1\end{array} \right)\right] X=-\left( \begin{array}{ccc}4 & 0 & -2\\1 & -3 & -1\end{array} \right)$

$\left[ \left( \begin{array}{cc}5 & 4\\3 & 8\end{array} \right)-\left(\begin{array}{cc}3 & 0\\0 & 3\end{array} \right)\right] X=\left( \begin{array}{ccc}-4 & 0 & 2\\-1 & 3 & 1\end{array} \right)$

$\left( \begin{array}{cc}2 & 4\\3 & 5\end{array} \right) X=\left( \begin{array}{ccc}-4 & 0 & 2\\-1 & 3 & 1\end{array} \right)$

Despejando..: $X=\left(\begin{array}{cc}2 & 4\\3 & 5\end{array} \right)^{-1}\left( \begin{array}{ccc}-4 & 0 & 2\\-1 & 3 & 1\end{array} \right)$

$X=\left(\begin{array}{cc}-2.5 & 2\\1.5 & -1\end{array} \right)\left( \begin{array}{ccc}-4 & 0 & 2\\-1 & 3 & 1\end{array} \right)$

$X=\left( \begin{array}{ccc}8 & 6 & -3\\-5 & -3 & 2\end{array} \right)$


$\textbf{Problema 1}$

$A=\left( \begin{array}{cc}2&2\\4&6\end{array} \right)$

Calcular la inversa de $A$.

$\textbf{Problema 2}$

suponemos que $A=\left( \begin{array}{cc}1&1\\0&0\end{array} \right) $

y que $B=\left( \begin{array}{cc}0&0\\2&0\end{array} \right) $

Calcula $A\cdot B$, y $B\cdot A$

$\textbf{Problema 3}$

suponemos que $A\cdot C = B$, y que $A=\left( \begin{array}{cc}1&3\\2&5\end{array} \right) $

y que $B=\left( \begin{array}{cc}0&1\\3&2\end{array} \right) $


Calcula C

$\textbf{Problema 4}$

Crear una función y guardarla con el nombre getPromedio, de tal forma que la función
reciba como argumento de entrada un Vector renglón y regrese como salida el promedio de
los datos.

$\textbf{Problema 5}$

Crear una función y guardarla con el nombre de getMCD, de tal forma que la función reciba como argumento de entrada 2 números, y regrese como salida el máximo común divisor de ambos.

El algoritmo de Euclides se basa en el hecho de que si $\textbf{a}$ es mayor que $\textbf{b}$, entonces el máximo común divisor de $\textbf{a}$ y $\textbf{b}$, es el mismo que de $\textbf{(a-b)}$ y $\textbf{b}$.

Si $\textbf{a}$ y $\textbf{b}$ son iguales, entonces ése es su valor de MCD.